Source/Core/VideoCommon/TextureConverterShaderGen.cpp

```diff
@@ -38,32 +38,81 @@ ShaderCode GenerateShader(APIType api_type, const UidData* uid_data)
   if (api_type == APIType::OpenGL)
   {
     out.Write("SAMPLER_BINDING(9) uniform sampler2DArray samp9;\n"
+              "uniform float3 filter_coefficients;\n"
+              "uniform float gamma_rcp;\n"
+              "uniform float2 clamp_tb;\n"
+              "uniform float pixel_height;\n"
               "#define samp0 samp9\n"
               "#define uv0 f_uv0\n"
               "in vec3 uv0;\n"
               "out vec4 ocol0;\n"
-              "void main(){\n"
-              "  vec4 texcol = texture(samp0, %s);\n",
-              mono_depth ? "vec3(uv0.xy, 0.0)" : "uv0");
+              "void main(){\n");
   }
   else if (api_type == APIType::Vulkan)
   {
+    out.Write("UBO_BINDING(std140, 1) uniform PSBlock {\n"
+              "  float3 filter_coefficients;\n"
+              "  float gamma_rcp;\n"
+              "  float2 clamp_tb;\n"
+              "  float pixel_height;\n"
+              "};\n");
     out.Write("SAMPLER_BINDING(0) uniform sampler2DArray samp0;\n"
               "layout(location = 0) in vec3 uv0;\n"
               "layout(location = 1) in vec4 col0;\n"
               "layout(location = 0) out vec4 ocol0;"
-              "void main(){\n"
-              "  vec4 texcol = texture(samp0, %s);\n",
-              mono_depth ? "vec3(uv0.xy, 0.0)" : "uv0");
+              "void main(){\n");
   }
   else if (api_type == APIType::D3D)
   {
     out.Write("Texture2DArray tex0 : register(t0);\n"
               "SamplerState samp0 : register(s0);\n"
+              "uniform float3 filter_coefficients;\n"
+              "uniform float gamma_rcp;\n"
+              "uniform float2 clamp_tb;\n"
+              "uniform float pixel_height;\n\n"
               "void main(out float4 ocol0 : SV_Target,\n"
               "          in float4 pos : SV_Position,\n"
-              "          in float3 uv0 : TEXCOORD0) {\n"
-              "  float4 texcol = tex0.Sample(samp0, uv0);\n");
+              "          in float3 uv0 : TEXCOORD0) {\n");
+  }
+
+  // Do not apply filter to depth copies.
+  // TODO: Verify this on hardware.
+  if (!uid_data->is_depth_copy)
+  {
+    // The filter is only applied to the RGB channels, the alpha channel is left intact.
+    if (api_type == APIType::OpenGL)
```

Comment:
Please define a small wrapper for the sampler in the header. So this code doesn't need to be forked for each backend.